modules/create_timepoint/php/ajaxtimepoint.class.inc

```diff
@@ -0,0 +1,300 @@
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class AjaxTimepoint extends \NDB_Page
+{
+    public $skipTemplate = true;
+
+    /**
+     * This function will return a json response.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        /**
+         * Ensure it's a POST request.
+         */
+        if ($request->getMethod() != 'POST') {
+            return (new \LORIS\Http\Response())
+                ->withHeader('Content-Type', 'application/json')
+                ->withStatus(405)
+                ->withHeader('Allow', 'POST')
+                ->withBody(
+                    new \LORIS\Http\StringStream(
+                        'Only POST requests are allowed'
+                    )
+                );
+        }
+        /**
+         * Parse POST request body.
+         */
+        $requestBody = $request->getParsedBody();
+
+        /**
+         * User permission verification.
+         */
+        $user = $request->getAttribute('user');
+        if (!isset($requestBody['identifier'])) {
+            throw new LorisException(
+                'Failed to retrieve data for create timepoint.'
+            );
+        }
+        $identifier = intval($requestBody['identifier']);
+        $candidate  = \Candidate::singleton($identifier);
+        if (!$user->hasPermission('data_entry')
+            && !(in_array(
+                intval($candidate->getData('RegistrationCenterID')),
+                $user->getData('CenterIDs'),
+                true
+            ))
+        ) {
+            return (new \Loris\Http\Response())
+                ->withStatus(403, 'Forbidden')
+                ->withBody(
+                    new \LORIS\Http\StringStream(
+                        json_encode(
+                            array('error' => 'Forbidden')
+                        )
+                    )
+                );
+        }
+        /**
+         * Sanitize $postResponse data.
+         */
+        $requestBody = $this->sanitizer($requestBody);
+
+        /**
+         * Response to reply.
+         */
+        $response = $this->processRequest($requestBody);
+        return (new \LORIS\Http\Response())
+            ->withHeader('Content-Type', 'application/json')
+            ->withStatus(200)
+            ->withHeader('Allow', 'POST')
+            ->withBody(
+                new \LORIS\Http\StringStream(
+                    json_encode($response)
+                )
+            );
+    }
+
+    /**
+     * Sanitize the $postResponse data.
+     *
+     * @param array $values that have been sanitized.
+     *
+     * @return array
+     */
+    function sanitizer(array $values) : array
```

Comment:
@johnsaigle I'm a little confused how security has dibs on the word sanitizer. I typically use that word when writing a method that confirms if the receiving data is actually valid; which the function her is checking if anything is set and or make it an int.

Reply:
This function was originally doing HTML-escaping of data that would be printed to the front-end. This is done in order to prevent from JavaScript injection attacks. That was the meaning behind the `$sanitizedIdentifier` in the old code. It's not that security has dibs but that you've changed what this function is doing from previous commits.

The discussion about the HTML-escaping not being necessary anymore seems to have been lost with all the commits, but that's the context I'm coming at this from.

It's misleading to me to still have this function given that we do use the word sanitize to mean security-checks in an informal context.

Even without that though, you don't need this function. You're already using `identifer` without any check on line 78. Also both your `validate()` and `initializeSetup` do different ways of validating `psc` so you are duplicating your validation code a few times within this file. 

So given that it's potentially misleading and also not really used, I think it makes sense to delete this function.

Reply:
@johnsaigle Thanks for the in-depth writeup. Yah when I look at the code it seems weird that I didn't place the function call of validating the response data before the permissions check in any case you're right that the function seems pointless. I'll just make some tweaks to storing the identifier as an int when I do the permissions check and delete the function. Hopefully nothing breaks.